src/formats/avroFormat.ts

```diff
@@ -0,0 +1,69 @@
+import { SchemaRegistry, SchemaType } from "@kafkajs/confluent-schema-registry";
+import { Env } from "../utils/env";
+import { OutputFormat } from "./outputFormat";
+
+export class AvroFormat implements OutputFormat {
+    private schemas: any = {};
+    private registry: SchemaRegistry;
```

Comment:
I liked having schema registry config decoupled from avro in case we can support protobuf or json_sr in the future, but I this is probably fine

Reply:
Those formats can also have a `SchemaRegistry` member. It can even reuse the same class. But passing in `record` and `recordEncoded` with lots of if statements interspersed is not maintainable.